src/main.cpp

```diff
@@ -0,0 +1,1605 @@
+
+        framebuffer_info.sType = VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO;
+        framebuffer_info.renderPass = render_pass;
+        framebuffer_info.attachmentCount = 1;
+        framebuffer_info.pAttachments = attachments;
+        framebuffer_info.width = swapchain_extent.width;
+        framebuffer_info.height = swapchain_extent.height;
+        framebuffer_info.layers = 1;
+
+        result = vkCreateFramebuffer(
+                logical_device,
+                &framebuffer_info,
+                nullptr,
+                &framebuffers[i]);
+        if (VK_SUCCESS != result) {
+            throw std::runtime_error("Failed to create framebuffer");
+        }
+    }
+
+    return framebuffers;
+}
+
+static VkCommandPool
+create_command_pool(
+        const VkDevice& logical_device,
+        const struct QueueFamilyIndices& family_indices)
+{
+    VkResult result;
+    VkCommandPool command_pool;
+    VkCommandPoolCreateInfo pool_info;
+
+    result = VK_ERROR_UNKNOWN;
+    pool_info = {};
+
+    pool_info.sType = VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
+    pool_info.flags = VK_COMMAND_POOL_CREATE_RESET_COMMAND_BUFFER_BIT;
+    pool_info.queueFamilyIndex = family_indices.drawing_family;
+
+    result = vkCreateCommandPool(
+            logical_device,
+            &pool_info,
+            nullptr,
+            &command_pool);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to create command pool");
+    }
+
+    return command_pool;
+}
+
+static VkCommandBuffer
+create_command_buffer(
+        const VkDevice& logical_device,
+        const VkCommandPool& command_pool)
+{
+    VkResult result;
+    VkCommandBuffer command_buffer;
+    VkCommandBufferAllocateInfo allocate_info;
+
+    result = VK_ERROR_UNKNOWN;
+    allocate_info = {};
+
+    allocate_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+    allocate_info.commandPool = command_pool;
+    allocate_info.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+    allocate_info.commandBufferCount = 1;
+
+    result = vkAllocateCommandBuffers(
+            logical_device,
+            &allocate_info,
+            &command_buffer);
+    if (VK_SUCCESS != result) {
+        throw std::runtime_error("Failed to allocate command buffers");;
+    }
+
+    return command_buffer;
+}
+
+static void
+record_command_buffer(
+        VkCommandBuffer command_buffer,
+        uint32_t image_index,
+        VkRenderPass& render_pass,
+        std::vector<VkFramebuffer>& swapchain_framebuffers,
+        VkExtent2D& swapchain_extent,
+        VkPipeline& graphics_pipeline)
+{
+    VkResult result;
+    VkCommandBufferBeginInfo begin_info;
+    VkRenderPassBeginInfo render_pass_info;
+    VkClearValue clear_color;
+    VkViewport viewport;
+    VkRect2D scissor;
+
+    result = VK_ERROR_UNKNOWN;
+    scissor = {};
+    viewport = {};
+    begin_info = {};
+    render_pass_info = {};
+    clear_color = {{{0.0f, 0.0f, 0.0f, 1.0f}}};
```

Comment:
Could single brackets be used here?

Reply:
Changed in: https://github.com/pottumuusi/neotetris/pull/1/commits/3e52caf15983b616c90201df55341a3002dbace2